tests/e2e/vcp_to_csi_attach_detach.go

```diff
@@ -0,0 +1,721 @@
+
+			framework.Logf("Deleting PVC %v", pvc.Name)
+			err = client.CoreV1().PersistentVolumeClaims(namespace).Delete(ctx, pvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		var defaultDatastore *object.Datastore
+		esxHost := GetAndExpectStringEnvVar(envEsxHostIP)
+		for _, pv := range pvsToDelete {
+			if pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimRetain {
+				err = client.CoreV1().PersistentVolumes().Delete(ctx, pv.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				if defaultDatastore == nil {
+					defaultDatastore = getDefaultDatastore(ctx)
+				}
+				if pv.Spec.CSI != nil {
+					err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					err = e2eVSphere.deleteFCD(ctx, pv.Spec.CSI.VolumeHandle, defaultDatastore.Reference())
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				} else {
+					if kcmMigEnabled {
+						found, crd := getCnsVSphereVolumeMigrationCrd(ctx, pv.Spec.VsphereVolume.VolumePath)
+						gomega.Expect(found).To(gomega.BeTrue())
+						err = e2eVSphere.waitForCNSVolumeToBeDeleted(crd.Spec.VolumeID)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						err = e2eVSphere.deleteFCD(ctx, crd.Spec.VolumeID, defaultDatastore.Reference())
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					}
+					err = deleteVmdk(esxHost, pv.Spec.VsphereVolume.VolumePath)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+			if pv.Spec.CSI != nil {
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			} else {
+				err = waitForVmdkDeletion(ctx, pv.Spec.VsphereVolume.VolumePath)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}
+
+		for _, crd := range crds {
+			framework.Logf("Waiting for CnsVSphereVolumeMigration crd %v to be deleted", crd.Spec.VolumeID)
+			err = waitForCnsVSphereVolumeMigrationCrdToBeDeleted(ctx, crd)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		vcpPvsPreMig = nil
+		vcpPvsPostMig = nil
+
+		if kcmMigEnabled {
+			err = toggleCSIMigrationFeatureGatesOnKubeControllerManager(ctx, client, false)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		vmdksToDel := vmdks
+		vmdks = nil
+		for _, vmdk := range vmdksToDel {
+			err = deleteVmdk(esxHost, vmdk)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		var scsToDelete []*storagev1.StorageClass
+		scsToDelete = append(scsToDelete, vcpScs...)
+		vcpScs = []*storagev1.StorageClass{}
+		for _, vcpSc := range scsToDelete {
+			err := client.StorageV1().StorageClasses().Delete(ctx, vcpSc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+	})
+
+	/*
+		Tests 1, 2, 3, 4, 7, 8, 13, 14, 15, 17, 18, 19, 20 from the TDS
+	*/
+	ginkgo.It("Attach detach combined test", func() {
```

Comment:
Internal links may not be ideal for the opensource code - please add one-liners for the tests

Reply:
If it becomes too lengthy to attach all the test steps. We can generalize the steps by giving some points on Whats the pre-data created and what all the validation points
Before migration : 
    <pre-data on SC,PVC,pod etc..>
After migration:
    <Validation Points>

Reply:
I've added the test case name from TDS.